### PR TITLE
Add timestamps property to favorites and add unique constraint to isbn…

### PR DESCRIPTION
**Pull request by cagallo**
… property

## What Features are Being Added?

- Add unique constraint to isbn property in favorites table
- Add timestamp property to favorites table

## What is Being Refactored?

- n/a 

## Where should the reviewer start?

- db/migrations

## How should this be manually tested?

- POST to favorites table
- view updated table 

## What are the relevant tickets?

- Add unique constraint to isbn and add timestamp property in favorites

## Any Other Notes/Questions/Problems?

- None at the moment

### CODE REVIEW

File: db/migrations/20220220213655_add_unique_constraint_isbn.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function(knex) {
+  return knex.schema.alterTable('favorites', (table) => {
+    table.unique('isbn')
```
Comment (Fordo29) [inline db/migrations/20220220213655_add_unique_constraint_isbn.js:7]: Good find!  I am so happy this worked in preventing multiple of the same to be added to a table

File: db/migrations/20220220213655_add_unique_constraint_isbn.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
```
Comment (krdurrett) [inline db/migrations/20220220213655_add_unique_constraint_isbn.js:4]: This is an excellent error handling approach to make sure we only have one copy of each book in favorites

File: db/seeds/books_favorites.js [hunk 3]
```diff
@@ -2,8 +2,8 @@
  * @param { import("knex").Knex } knex
  * @returns { Promise<void> } 
  */
-const booksData = require('../data/filtered-books-data');
-const favorites = require("../data/favorites-data");
+const booksData = require('../../data/filtered-books-data');
+const favorites = require("../../data/favorites-data");
```
Comment (ectweitmann) [inline db/seeds/books_favorites.js:6]: Nice catch. I missed updating these file paths when I created the db folder.

File: db/migrations/20220220215434_create_timestamp_property_favorites.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @param { import("knex").Knex } knex
+ * @returns { Promise<void> }
+ */
+exports.up = function(knex) {
+  return knex.schema.table('favorites', (table) => {
+    table.timestamps(true, true);
+  });
```
Comment (krdurrett) [inline db/migrations/20220220215434_create_timestamp_property_favorites.js:8]: Good idea to add timestamp to favorites as well.

**Review by krdurrett — APPROVED** [review]
Great changes to the favorites table!

File: db/migrations/20220220213655_add_unique_constraint_isbn.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+exports.up = function(knex) {
+  return knex.schema.alterTable('favorites', (table) => {
+    table.unique('isbn')
+  })
```
Comment (ectweitmann) [inline db/migrations/20220220213655_add_unique_constraint_isbn.js:8]: This is a cool way to keep the favorites array free of duplicate books!

**Review by ectweitmann — APPROVED** [review]

**Pull request merged**